query-node/mappings/src/content/nft.ts

```diff
@@ -930,3 +945,44 @@ export async function contentNft_BuyNowCanceled({ event, store }: EventContext &
 
   await store.save<BuyNowCanceledEvent>(announcingPeriodStartedEvent)
 }
+
+export async function contentNft_NftSlingedBackToTheOriginalArtist({
+  event,
+  store,
+}: EventContext & StoreContext): Promise<void> {
+  // common event processing
+
+  const [videoId, ownerId] = new Content.NftBoughtEvent(event).params
+
+  // load NFT
+  const { video, nft } = await getNftFromVideo(
+    store,
+    videoId.toString(),
+    'Non-existing video was slinged',
+    'Non-existing nft was slinged'
+  )
+
+  // load channel
+  const channel = (await getById(store, Video, video.getId(), ['channel'])).channel
+
+  // load channel ownerCuratorGroup (if any)
+  const ownerCuratorGroup = (await getById(store, Channel, channel.getId(), ['ownerCuratorGroup'])).ownerCuratorGroup
```

Comment:
This can be simplified with relation dependency loading. FYI: relations can be chained e.g. `['channel', 'channel.ownerCuratorGroup', 'channel.ownerCuratorGroup.whateverRelationExistsThere']`
```suggestion
  const { video, nft } = await getNftFromVideo(
    store,
    videoId.toString(),
    'Non-existing video was slinged',
    'Non-existing nft was slinged',
    ['channel', 'channel.ownerCuratorGroup', 'channel.ownerMember']
  )
```